model/project_ref.go

```diff
@@ -1791,6 +1792,10 @@ func SaveProjectPageForSection(projectId string, p *ProjectRef, section ProjectP
 			setUpdate[ProjectRefOwnerKey] = p.Owner
 			setUpdate[ProjectRefRepoKey] = p.Repo
 		}
+		if !defaultToRepo {
```

Comment:
Can we add a comment here to explain why this section is different? (honestly I should've done above too)